### PR TITLE
Simplify LLM inference modeling and use longer sequences in tests

### PR DESCRIPTION
**Pull request by dacorvo**
# What does this PR do?

This pull request refactors the decoder module input handling and attention mask logic to simplify tensor management and remove the need for explicit `attention_mask` arguments in most decoder flows. The changes streamline padding and batching logic, update model interfaces, and consolidate attention mask creation within the model graph itself, improving code clarity and maintainability.

The cached models test fixtures is also revamped, and test configurations using longer sequences are added.

### CODE REVIEW

File: optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py [hunk 1]
```diff
@@ -168,28 +142,46 @@ def forward(
 
         cache_size = self.n_positions
 
-        # It is either for context encoding or for token generation
+        # Prepare input tensors
+        device = input_ids.device
         if is_for_context_encoding:
             past_key_values = None
+            # Lower triangle causal mask for classic attention
```
Comment (tengomucho) [inline optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py:148]: perhaps not now, but I think at some point we might want to isolate this to handle sliding window attention too.

File: tests/decoder/test_decoder_generation.py [hunk 2]
```diff
@@ -93,26 +93,28 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwa
 @is_inferentia_test
 @requires_neuronx
 def test_decoder_generation_greedy_expectations(neuron_llm_config):
+    model_id = neuron_llm_config["model_id"]
+    model = AutoModelForCausalLM.from_pretrained(model_id)
```
Comment (tengomucho) [inline tests/decoder/test_decoder_generation.py:99]: nit: consider renaming this `cpu_model` to make it clearer it runs on cpu

File: optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py [hunk 3]
```diff
@@ -416,51 +415,15 @@ def forward(
...
-                raise ValueError("The sampling params tensor is required for on-device sampling.")
-            # Just pass a dummy tensor to the model, it will be ignored
-            sampling_params = prepare_sampling_params(seq_ids.shape[0])
-        elif self.neuron_config.on_device_sampling:
+        if output_attentions:
+            raise ValueError(f"output_attentions is not supported for {self.__class__.__name__}")
+        if output_hidden_states:
+            raise ValueError(f"output_hidden_states is not supported for {self.__class__.__name__}")
+        if return_dict:
+            raise ValueError(f"return_dict is not supported for {self.__class__.__name__}")
```
Comment (tengomucho) [inline optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py:423]: couldn't we remove these params instead of leaving them and raise an error when used?

File: optimum/neuron/models/inference/backend/modules/decoder/decoder_builders.py [hunk 4]
```diff
@@ -61,8 +61,16 @@ def input_generator(
         inputs = []
 
         input_ids = torch.zeros((self.neuron_config.batch_size, self.active_tokens), dtype=torch.int32)
-        position_ids = torch.zeros((self.neuron_config.batch_size, self.active_tokens), dtype=torch.int32)
-        seq_ids = torch.zeros((self.neuron_config.batch_size), dtype=torch.int32)
+        if self.active_tokens == self.max_tokens:
+            # Context encoding
+            position_ids = (
+                torch.arange(self.active_tokens, dtype=torch.int32)
+                .unsqueeze(0)
```
Comment (tengomucho) [inline optimum/neuron/models/inference/backend/modules/decoder/decoder_builders.py:68]: I think you do not need the `unsqueeze(0)` here, since you expand it right away.

File: optimum/neuron/models/inference/backend/modules/decoder/decoder_builders.py [hunk 5]
```diff
@@ -145,7 +153,11 @@ def input_generator(
     ):
         input_ids = torch.zeros((self.neuron_config.batch_size, self.max_tokens), dtype=torch.int32)
         attention_mask = torch.zeros((self.neuron_config.batch_size, self.max_tokens), dtype=torch.int32)
-        position_ids = torch.zeros((self.neuron_config.batch_size, self.max_tokens), dtype=torch.int32)
+        position_ids = (
+            torch.arange(self.max_tokens, dtype=torch.int32)
+            .unsqueeze(0)
```
Comment (tengomucho) [inline optimum/neuron/models/inference/backend/modules/decoder/decoder_builders.py:158]: same here, I think `unsqueeze` is not needed.

File: optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py (hunk 3 above)
Comment (dacorvo) [reply optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py:423]: They are accepted by `transformers`

File: optimum/neuron/models/inference/backend/modules/decoder/decoder_builders.py (hunk 4 above)
Comment (dacorvo) [reply optimum/neuron/models/inference/backend/modules/decoder/decoder_builders.py:68]: You cannot expand a 1D tensor to 2D, ... or can you 🤔 ?

Comment (tengomucho) [reply optimum/neuron/models/inference/backend/modules/decoder/decoder_builders.py:68]: yes you can, e.g.:
```python
In [1]: import torch

In [2]: torch.arange(4).expand(2, 4)
Out[2]:
tensor([[0, 1, 2, 3],
        [0, 1, 2, 3]])
```

File: optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py (hunk 3 above)
Comment (tengomucho) [reply optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py:423]: but do we accept all params transformers accepts?

Comment (dacorvo) [reply optimum/neuron/models/inference/backend/modules/decoder/modeling_decoder.py:423]: Correction: they are passed by `transformers` with a value of None

**Review by tengomucho — APPROVED** [review]
